### PR TITLE
inlets 0.6.3 (new formula) 

### PR DESCRIPTION
**Pull request by johnmccabe**
Project:
Expose your local endpoints to the Internet

CLI: inlets

2k GitHub stars

Tested locally with brew test.

Signed-off-by: Alex Ellis alexellis2@gmail.com
Signed-off-by: John McCabe john@johnmccabe.net

*NOTE* This PR supercedes #37420 and adds a non-trivial test of the client where the ping-pong keep alive messaging is tested between the client and a mock server.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by johnmccabe** [comment]
@fxcoudert @alexellis this PR replaces #37420 

Let me know if the new test is good enough.

I'd ideally have liked to fork directly rather than run the mock server via the forked exec, but the handover to ephemeral ports fails when I fork from within the brew formula - it *seems* to be related to the `sandbox-exec` as the mock server works as expected when split out into the `ws_server.rb` file.

**Comment by johnmccabe** [comment]
Fixing the audit comments (missed the output).

**Comment by johnmccabe** [comment]
Fixed audit warnings.

```
…/homebrew-core  inlets_websocket_test # brew test inlets
Testing inlets
Listening on: 63068
==> /usr/local/Cellar/inlets/0.6.2/bin/inlets --version
==> /usr/local/Cellar/inlets/0.6.2/bin/inlets client -r 127.0.0.1:63068 -u http://127.0.0.1:8080 -p 1s 2>&1
Incoming Request
Websocket handshake detected with key: lJJfgUVD8OQ8nAR4GeBgvw==
Responding to handshake with key: Uk8v5ikelEJqJJ4sBZhK4nMJbh8=
Handshake completed. Starting to parse the websocket frame.
Received Ping
Sending Pong
Received Ping
Sending Pong
Exiting websocket server

…/homebrew-core  inlets_websocket_test # brew audit --strict inlets

```

**Comment by johnmccabe** [comment]
Anything else I can do to help this along.

File: Formula/inlets.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  desc "Expose your local endpoints to the Internet"
+  homepage "https://github.com/alexellis/inlets"
+  url "https://github.com/alexellis/inlets.git",
+      :tag      => "0.6.2",
+      :revision => "fbc9f4fe98cf48e44300736140bd013d64d6f980"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["XC_ARCH"] = "amd64"
```
Comment (SMillerDev) [inline Formula/inlets.rb]: Is this required for the formula to work?

Comment (johnmccabe) [reply Formula/inlets.rb]: If you mean setting the Arch, then I don't believe so. It should pick that up from the build env

Comment (SMillerDev) [reply Formula/inlets.rb]: Could you remove it then?

**Comment by johnmccabe** [comment]
@SMillerDev I've purged the `XC_ARCH` and have re-run the install/test/audit steps successfully.

**Comment by idcrook** [comment]
would it make sense to be bumped to version `0.6.3`?

**Comment by johnmccabe** [comment]
Was holding off on merge rather than introduce churn at this point.

**Comment by SMillerDev** [comment]
Well, I'm not going to merge outdated software into homebrew. Can you bump to the latest version?

**Comment by johnmccabe** [comment]
@SMillerDev bumped version and have re-run the install/test/audit steps successfully.

**Comment by fxcoudert** [comment]
Thanks @johnmccabe for the pull request! Merging

**Pull request closed without merging**